### PR TITLE
Merge 'main' into 'develop' and update version number to 1.1.0

### PR DESCRIPTION
**Pull request by Nox-MSFT**
- Merging 'main' into 'develop'
- Update the agent version to 1.1.0

### CODE REVIEW

File: scripts/install-deps.sh [hunk 1]
```diff
@@ -527,7 +527,7 @@ do_install_azure_storage_sdk() {
 
     cmake "${azure_storage_sdk_cmake_options[@]}" . || return
 
-    cmake --build . || return
+    cmake --build . || return 1
```
Comment (nihemstr) [inline scripts/install-deps.sh:530]: nit: Change to common exit pattern unless there's reasons to have it return 1

File: src/inc/aduc/result.h [hunk 2]
```diff
@@ -2119,6 +2124,21 @@ MAKE_ADUC_EXTENDEDRESULTCODE_FOR_FACILITY_ADUC_FACILITY_UNUSED_F(const int32_t c
 #define ADUC_ERC_DOWNLOAD_HANDLER_EXTENSIBILITY_GET_CONTRACT \
     MAKE_ADUC_EXTENDEDRESULTCODE_FOR_COMPONENT_ADUC_COMPONENT_DOWNLOAD_HANDLER_EXTENSION_MANAGER(5)
 
+/**
+ * @brief ADUC_ERC_DOWNLOAD_HANDLER_CREATE_FACTORY_INSTANCE, ERC Value: 2415919107 (0x90000003)
+ */
+ #define ADUC_ERC_DOWNLOAD_HANDLER_CREATE_FACTORY_INSTANCE MAKE_ADUC_EXTENDEDRESULTCODE_FOR_COMPONENT_ADUC_COMPONENT_DOWNLOAD_HANDLER_EXTENSION_MANAGER(3)
```
Comment (jw-msft) [inline src/inc/aduc/result.h]: Remove? DUP of line 2112-2125.
Probably better off building clean and doing git add on generated result.h

File: src/adu_workflow/src/agent_workflow.c [hunk 3]
```diff
@@ -1208,6 +1209,8 @@ static void CallDownloadHandlerOnUpdateWorkflowCompleted(const ADUC_WorkflowHand
                 workflow_add_erc(workflowHandle, result.ExtendedResultCode);
             }
         }
+
+        ADUC_FileEntity_Uninit(&fileEntity);
```
Comment (nihemstr) [inline src/adu_workflow/src/agent_workflow.c:1213]: Line 1198 is already uninitializing the fileEntity.

File: scripts/install-deps.sh (hunk 1 above)
Comment (jw-msft) [inline scripts/install-deps.sh:530]: Is it better to overwrite (with 1) the cmake exit code on failure? line 531 for instance does not. Which commit in main is this from?

**Review by nihemstr — APPROVED** [review]
Approved with comments

File: azurepipelines/build/docker/adu-debian-arm32-build.yml [hunk 4]
```diff
@@ -20,6 +27,13 @@ variables:
     # gcc-8 matches what is built with poky warrior.
     CC: gcc-8
     CXX: g++-8
+    ENABLE_ADU_TELEMETRY_REPORTING: true
+    # ADUC_BUILDER_IDENTIFIER will be set to "DU" short for Device Update by default, for Device Update-sourced builder
+    ADUC_BUILDER_IDENTIFIER: DU
+    # DO requires gcc greater than 6 for c++17 support.
+    # gcc-8 matches what is built with poky warrior.
+    CC: gcc-8
```
Comment (jw-msft) [inline azurepipelines/build/docker/adu-debian-arm32-build.yml]: seems like it's DUP of lines 28-29?

File: scripts/install-deps.sh (hunk 1 above)
Comment (Nox-MSFT) [reply scripts/install-deps.sh]: Fixed

Comment (Nox-MSFT) [reply scripts/install-deps.sh]: Fixed

File: src/adu_workflow/src/agent_workflow.c (hunk 3 above)
Comment (Nox-MSFT) [reply src/adu_workflow/src/agent_workflow.c]: fixed

**Review by Jerry-GeGarcia — APPROVED** [review]

File: src/utils/system_utils/src/system_utils.c [hunk 5]
```diff
@@ -19,6 +19,7 @@
 
 #include <aduc/string_c_utils.h>
 #include <azure_c_shared_utility/strings.h>
+#include <dirent.h>
```
Comment (jw-msft) [inline src/utils/system_utils/src/system_utils.c:22]: Likely can remove this line. I believe it's causing windows build pipeline to fail:

https://dev.azure.com/azure-device-update/adu-linux-client/_build/results?buildId=14245&view=logs&j=2dd7e357-93c9-5cd9-3155-fb06f1278828&t=185a2715-b6bc-50a7-3503-0bdaa3b94916&l=744

D:\a\_work\1\s\src\utils\system_utils\src\system_utils.c(22,10): fatal  error C1083: Cannot open include file: 'dirent.h': No such file or directory [D:\a\_work\1\s\out\Debug\src\utils\system_utils\system_utils.vcxproj]

ERROR: Build failed (Error 1)

File: src/inc/aduc/result.h (hunk 2 above)
Comment (nihemstr) [reply src/inc/aduc/result.h]: How is this getting duped? It should be generated by the code never touched by us. There should be no way for there to be a duplicate

File: src/utils/system_utils/src/system_utils.c (hunk 5 above)
Comment (nihemstr) [reply src/utils/system_utils/src/system_utils.c]: Other option is to add this to libaducpal to handle management.

File: src/inc/aduc/result.h (hunk 2 above)
Comment (jw-msft) [reply src/inc/aduc/result.h]: > How is this getting duped? It should be generated by the code never touched by us. There should be no way for there to be a duplicate
@nihemstr maybe it was somehow a bad merge of checked-in result.h, which is why I mentioned to regenerate it.

**Review by jw-msft — APPROVED** [review]

**Pull request merged**